state/core_access.go

```diff
@@ -218,6 +226,14 @@ func (ca *CoreAccessor) SubmitPayForBlob(
 
 	minGasPrice := ca.getMinGasPrice()
 
+	feeGrant, err := ca.getGranter(ctx)
+	if err != nil {
+		log.Warn(err)
+	}
```

Comment:
This adds "yet another roundtrip" to the tx submission. Is there a way to persisting (and pre-loading) that info locally instead of quering remote address?

cc @vgonkivs and @cmwaters

Reply:
Also, shouldn't feeGranter be supported for all tx submissions and not just blobs?

Reply:
Yeah I agree, if a user knows they have a grant then maybe they can simply initiate the client with the granter address (i.e. in the constructor) and we do one query at the start to see what messages have grants and even if it exists and then use that address for every message

Reply:
How do users know when the granter revokes the grant in this case? I guess we will anyway need to do a roundtrip.

Reply:
Through an error. You can do it once you start the node, and if, during runtime, the granter revokes - the transaction will error. This works, and we can recommend that users restart the node to fix the error. Alternatively, we can spend some time making custom error-handling logic for this case that removes granting.